src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Figure;
+import core.basesyntax.figures.FigureSupplier;
+
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
+        for (int i = 0; i < figures.length; i++) {
+            FigureSupplier figureSupplier = new FigureSupplier();
+            ColorSupplier colorSupplier = new ColorSupplier();
```

Comment:
dont create FigureSupplier object inside for-loop - you are creating unnecessary instances
you don't need ColorSupplier in main - you should use it inside the FigureSupplier